Stop handling exceptions

cintrzyk opened the pull request:
Automatically rescuing all exceptions can obscure unexpected errors and complicate debugging. Not catching exceptions by default gives developers greater control over the execution flow, making failures more transparent. Similar libraries, such as Trailblazer, also avoid implicit exception handling.

msx2 commented on spec/opera/operation/base_spec.rb:
```diff
@@ -405,31 +405,45 @@ def step_1
+            end
+          end
+
+          it 'does not handle exception' do
```
Maybe consistent
```suggestion
          it 'raises exception' do
```

as you used in L445?

msx2 commented on spec/opera/operation/result_spec.rb:
```diff
@@ -19,11 +19,10 @@ module Opera
 
-      it 'returns errors and exceptions combined' do
-        expect(subject.failures).to eq(foo1: [:bar1], 'foo2' => :bar2)
+      it 'returns errors combined' do
```
We're not combining anymore :)

```suggestion
      it 'returns errors' do
```

msx2 reviewed (approved):
🚀

cintrzyk commented:
@petergebala @kikorb I had to specify 9.4 version of jruby which [is point release of Ruby 3.1.x support](https://www.jruby.org/download). While doing it, I've also updated some gems and dropped support for ruby 3.0 
<img width="640" alt="image" src="https://github.com/user-attachments/assets/baa1b10e-0dfe-4de5-9f70-c97fdb3847b0" />
